src/core/createCookie.js

```diff
@@ -22,55 +22,47 @@ const createCookie = (prefix, id, cookieDomain = "") => {
      */
     get(name) {
       const cookieName = `${ALLOY_COOKIE_NAME}_${id}`;
-      let currentCookie;
-      try {
-        currentCookie = cookie.getJSON(cookieName);
-        return currentCookie[prefix][name];
-      } catch (error) {
-        return currentCookie;
-      }
+      const currentCookie = cookie.get(cookieName);
+      const currentCookieParsed = currentCookie && JSON.parse(currentCookie);
+      return (
+        currentCookieParsed &&
+        currentCookie[prefix] &&
```

Comment:
Shouldn't these be `currentCookieParsed[prefix] && currentCookieParsed[prefix][name]`?

Doesn't `JSON.parse` throw if `currentCookie` is `undefined` or an empty string? I think we still need to `try / catch` the `JSON.parse` expressions. You can even create a small helper, `safeJsonParse` that does the wrapping for you.

Reply:
Yes you are right, it should be `currentCookieParsed`
It wouldn't go to `JSON.parse` if `currentCookie` is `undefined`.